File: ScaleObj.py
"""
    Created on Fri Nov 17 22:34:05 2017
    
    @author: akshayjaitly
    """
import sys

# Inputs must be in the form filename.obj x y z
# Steps to run : python ScaleObj.py mini.obj 2 3 2
if (len(sys.argv) != 5):
    print("Incorrect no. of input arguments, Inputs must be : filename x y z")

# Declaring arguements
infile = sys.argv[1]
x_scale = sys.argv[2]
y_scale = sys.argv[3]
z_scale = sys.argv[4]

# Parse the x,y,z to float
x_scale = float(x_scale)
y_scale = float(y_scale)
z_scale = float(z_scale)

# Declaring input and output paths
file_in = open(infile, "r")
file_out = open("Scale"+ infile, "w")
# Read all lines in file using readline
for s in file_in.readlines():
    output = s
    tmp = s.split() # Split string into list
    if(len(tmp) > 0 and tmp[0] == "v" ):
        if(len(tmp) != 4):
            print("Fault! .obj file has vertices with more than 3 axes ..." + tmp)
            continue
            
        # x axis
        x_axis = float(tmp[1])
        x_axis *= x_scale
        # y axis
        y_axis = float(tmp[2])
        y_axis *= y_scale
        # z axis
        z_axis = float(tmp[3])
        z_axis *= z_scale
        
        # Output which is to be displayed using string concatenation
        output = tmp[0] + " " + str(x_axis) + " " + str(y_axis) + " " + str(z_axis) + "\n"

    # Surface normals are skipped
    if(len(tmp) > 0 and tmp[0] == "vn"):
        continue

    # Surface tangents are skipped
    if(len(tmp) > 0 and tmp[0] == "vt"):
      continue

    # Writing to output
    file_out.write(output)

File: ScaleObj.java
/* Akshay Jaitly Cs 635 Programming Assignment
Steps to run program in Java:
javac ScaleObj.java
java ScaleObj Sample.obj 2 1 16
output Filename:Scale.obj */
import java.io.*;

public class ScaleObj {
	public static void main(String[] args){
		if(args.length != 4){
			System.err.println("Wrong no. of input arguments, Inputs must be : filename.obj x y z");
			return;
		}
		
		// Declare arguemnts
		String filename = args[0];//arguement for filename
		float x_scale;//Scaling factor for x
		float y_scale;//Scaling factor for y
		float z_scale;//Scaling factor for z
		try{
			x_scale = Float.parseFloat(args[1]);//Float conversion
			y_scale = Float.parseFloat(args[2]);//Float conversion
			z_scale = Float.parseFloat(args[3]);//Float conversion
		}catch(Exception e){
			System.err.println("Error parsing the user inputted scaling factors");
			return;
		}

		String line = null;
        
		try{
			// Creating output file from an input file
			File f = new File(filename);
            String outputfile = "Scale.obj" ;//Scaled Output file
			System.out.println("Generating Output File : " + outputfile);
            
			// Reading from input file
			FileReader fileReader = new FileReader(filename);
			BufferedReader bufferedReader = new BufferedReader(fileReader);
			
			// Writing to output file
			FileWriter fileWriter = new FileWriter(outputfile);
			BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);			
			
			while((line = bufferedReader.readLine()) != null){
				// Output line to write into output file
				String output = line;
				
				String[] splitline = line.split("\\s+");
				if("v".equalsIgnoreCase(splitline[0])){
					if(splitline.length != 4){
						System.err.println("Fault! Obj file contains vertices with more than 3 coordinates...");
						System.err.println(line);						
					}
					
					// x axis
					float x_axis = Float.parseFloat(splitline[1]);
					x_axis *= x_scale;
					// y axis
					float y_axis = Float.parseFloat(splitline[2]);
					y_axis *= y_scale;
					// z axis
					float z_axis = Float.parseFloat(splitline[3]);
					z_axis *= z_scale;
					
                    // Output using string concatenation
                    output = splitline[0] + " " + Float.toString(x_axis);
                    output += " " + Float.toString(y_axis);
                    output += " " + Float.toString(z_axis);
				}
				
                // Ignore surface normals if any
				if("vn".equalsIgnoreCase(splitline[0])){
					continue;
				}
				
                // Ignore surface tangents if any
                if("vt".equalsIgnoreCase(splitline[0])){
                    continue;
                }
                
				bufferedWriter.write(output);
				bufferedWriter.newLine();
			}
			
			bufferedReader.close();
			bufferedWriter.close();
		}catch(FileNotFoundException ex){
			System.err.println("Cannot open file '" + filename + "'");
		}catch(IOException e){
			System.err.println("Error! Can't read file '" + filename + "'");
		}
		
		System.out.println("A scaled Wavefront .obj file is created!");
	}

}
